### ISSUE TITLE
Integrate GEDA and Gaussian Autoregressive models

### ISSUE THREAD

**Issue by cweniger** [undark-lab/swyft#140]
In order to make that happen, we need to consolidate the Gaussian_Autoregressive interface. What about dropping the `_X`, and only keeping the module-based version? We can later add functionality to default to linear modules for `L` and `Phi`.  Having two different versions of essentially the same function is probably not the best idea.

@NoemiAM, Florian, any thoughts about this?

**Comment by NoemiAM** [comment]
@cweniger I agree we should drop one of the two since they are basically the same. But the consolidated version should contain both sampling methods: standard Cholesky decomposition and GEDA.
